### PR TITLE
bpo-43897: Reject `"_"` captures and top-level `MatchStar` in the AST validator

### PR DESCRIPTION
**Pull request by brandtbucher**
Also, remove an old comment and fix some recursion depth issues.

https://bugs.python.org/issue43897

### CODE REVIEW

File: Python/ast.c [hunk 1]
```diff
@@ -493,16 +493,24 @@ validate_pattern_match_value(struct validator *state, expr_ty exp)
...
+{
+    if (_PyUnicode_EqualToASCIIString(name, "_")) {
+        PyErr_Format(PyExc_ValueError, "can't capture name '_' in patterns");
+        return 0;
+    }
+    return validate_name(name);
+}
+
+static int
+validate_pattern(struct validator *state, pattern_ty p, int star_okay)
```
Comment (isidentical) [inline Python/ast.c]:
```suggestion
validate_pattern(struct validator *state, pattern_ty p, int star_ok)
```

**Review by isidentical — APPROVED** [review]
LGTM (beside `star_ok`)

**Comment by isidentical** [comment]
> Generally, these issues must be severe enough (e.g. crashes) that they deserve fixing before the final release. All other issues should be deferred to the next development cycle (Python 3.10.1) since stability is the strongest concern at this point

Regarding the backport to 3.10, I think it might make sense to hold this off until `3.10.1` what do you think? Also we should probably notify pablo on the backport PR

File: Python/ast.c (hunk 1 above)
Comment (brandtbucher) [reply Python/ast.c]: Nice catch!

**Comment by brandtbucher** [comment]
That's only *after* the first RC, right? I think a backport for this would be fine as long as it happened soon.

@pablogsal?

**Comment by pablogsal** [comment]
> That's only *after* the first RC, right? I think a backport for this would be fine as long as it happened soon.
> 
> @pablogsal?

That's correct, that is only *after* the first RC. Please, merge this as soon as possible so we can test it enough before we release the RC.

 Ideally someone has a fuzzer powerful enough to check the validator 😉

**Comment by brandtbucher** [comment]
Windows (x86) failure looks unrelated.

**Pull request merged**